dask/dataframe/tests/test_shuffle.py

```diff
@@ -596,6 +597,19 @@ def test_set_index_sorted_min_max_same():
     assert df2.divisions == (0, 1, 1)
 
 
+def test_set_index_text_column_empty_partition():
+    df = pd.DataFrame([{'x': str(i), 'y': i} for i in range(3)], columns=['x', 'y'])
+    ddf = dd.concat([
+        dd.from_pandas(df, npartitions=1),
+        dd.from_pandas(df[df.y > df.y.max()], npartitions=1),
+    ])
+
+    assert any(ddf.get_partition(p).compute().empty for p in range(ddf.npartitions))
+
+    ddf = ddf.set_index('x')
+    assert assert_eq(ddf.index.compute(), pd.Index(['0', '1', '2'], dtype='object', name='x'))
```

Comment:
You shouldn't need to call `compute()` here.  Assert_eq will do that for you after checking metadata on the dask dataframe object itself.

Reply:
Can we call something like `assert_eq(ddf, df.set_index(...))` as well just for completeness?

Reply:
I'll remove the call to compute